lib/railway_ipc/ipc/consumers/republished_messages_consumer.ex

```diff
@@ -0,0 +1,30 @@
+defmodule RailwayIpc.Ipc.RepublishedMessagesConsumer do
+  use RailwayIpc.CommandsConsumer,
+    queue: "railway_ipc:republished_messages:commands",
+    publish_function: &RailwayIpc.Publisher.publish/1
+
+  alias RailwayIpc.Ipc.Logger
+  alias LearnIpc.Commands.RepublishMessage
+  alias RailwayIpc.PublishedMessage
+
+  def handle_in(
+        %RepublishMessage{data: %{published_message_uuid: published_message_uuid}} = message
+      ) do
+    case PublishedMessage.get(published_message_uuid) do
+      nil ->
+        Logger.log_message_handled_failure(
+          message,
+          "Unable to find published message with UUID: #{published_message_uuid}"
+        )
+
+        :ok
+
+      %{message_type: "LearnIpc::Commands::RepublishMessage"} ->
```

Comment:
This ensures that we don't re-publish messages of the "republish message command" type b/c otherwise o boy too much confusion. Also ensures we don't go through with publishing a message to a nil exchange which would just blow up weird.